firmware/src/sys/board.hpp

```diff
@@ -58,6 +58,9 @@ void resetWatchdog();
 
 void setStatusLed(bool state);
 void setCanLed(bool state);
+void DutycycleCounterPP();
+void DutycycleCounterMM();
+signed DutycycleCounterRead();
```

Comment:
Thank you Pavel, I'll get to work on this

On Thu, Aug 4, 2016 at 6:36 AM, Pavel Kirienko <notifications@github.com>
wrote:

> In firmware/src/sys/board.hpp
> <https://github.com/Zubax/opengrab_epm_v3/pull/33#discussion_r73513037>:
>
> > @@ -58,6 +58,9 @@ void resetWatchdog();
> >
> >  void setStatusLed(bool state);
> >  void setCanLed(bool state);
> > +void DutycycleCounterPP();
> > +void DutycycleCounterMM();
> > +signed DutycycleCounterRead();
>
> Thanks. All of the logic you described must be constrained to the magnet
> module. Right now it's spread out thinly over the entire project; while it
> could be considered acceptable in this particular case right now (the
> application is very simple), it will make the code unmaintainable and full
> of bugs in the medium and long term. What you need to do is this:
>
>    - Revert your changes to all files except magnet.cpp
>    - Define the counter variable as a module-local in magnet.cpp (around
>    here
>    <https://github.com/Zubax/opengrab_epm_v3/blob/master/firmware/src/magnet/magnet.cpp#L169>
>    )
>    - Make sure that the counter variable is appropriately typed (counters
>    like yours should be unsigned); if you need constants, they must be defined
>    constexpr.
>    - Decrement your counter in these two places:
>       -
>       https://github.com/Zubax/opengrab_epm_v3/blob/master/firmware/src/magnet/magnet.cpp#L189
>       -
>       https://github.com/Zubax/opengrab_epm_v3/blob/master/firmware/src/magnet/magnet.cpp#L225
>    - Increment your counter every 100 ms (ar as frequently as you need)
>    here:
>    https://github.com/Zubax/opengrab_epm_v3/blob/master/firmware/src/magnet/magnet.cpp#L301;
>    add a static variable to keep track of time.
>
> I want to emphasize again that this change must be constrained to one
> module only. The API of the magnet module need not be changed either.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Zubax/opengrab_epm_v3/pull/33/files/f79d22098656f8fcdedc07e55bb7cbd485e3b04a#r73513037>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFydXYSeD6h6aIaxLmTzx5OQ1Qflrxfcks5qcdzdgaJpZM4JcA4Z>
> .
>